applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -440,6 +449,16 @@ int cloud_wrap_cloud_location_send(char *buf, size_t len, bool ack, uint32_t id)
 	return 0;
 }
 
+bool cloud_wrap_cloud_location_response_wait(void)
+{
+	/* If GNSS is the first priority, then we can ignore the location response from the cloud */
+#if defined(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST_GNSS)
+	return false;
+#else
+	return true;
+#endif
```

Comment:
I've currently done this function so that cloud implementation can decide whether the response is being waited for or not. AWS and Azure integrations will just return false here always.

I've intentionally made here in nRF Cloud integration the trick to return false or true depending on `CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST_GNSS` to provoke discussion. Originally I had here just `return true`. Flagging on lines 239-245 is tightly related to this.

@simensrostad @tokangas Comments are welcome.

Reply:
Yes, this is the most "correct" implementation. However, things done in this PR to handle location response somewhere and ignore in other places complicates the implementation and doesn't make it easier for someone else to come and maintain the code :) It would be easier if we would ignore or handle it always irrespective of the used cloud backend and location method priority list ordering.

Reply:
Even with this implementation for WiFi+GNSS+Cellular priority order, we will wait for the location unnecessarily for the last Cellular but I'm not sure that's something we need to optimize.